packages/core-sdk/src/resources/ipAsset.ts

```diff
@@ -2450,4 +2465,19 @@ export class IPAssetClient {
     }
     return ipAsset;
   }
+
+  private async approveLicenseTokensForDerivativeWorkflows(
+    licenseTokenIds: LicenseTermsIdInput[],
+    autoApproveLicenseTokens?: boolean,
+  ): Promise<void> {
+    if (autoApproveLicenseTokens !== false) {
+      for (const licenseTokenId of licenseTokenIds) {
+        const txHash = await this.licenseTokenClient.approve({
+          to: this.derivativeWorkflowsClient.address,
+          tokenId: BigInt(licenseTokenId),
+        });
+        await waitTx(this.rpcClient, txHash);
```

Comment:
Also, instead of waiting on one transaction at a time, can we send the transactions and wait all at once (using promises)?